optimum/deepsparse/modeling_diffusion.py

```diff
@@ -0,0 +1,527 @@
+#  Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import importlib
+import logging
+import os
+import shutil
+from abc import abstractmethod
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from diffusers import (
+    DDIMScheduler,
+    LMSDiscreteScheduler,
+    PNDMScheduler,
+    StableDiffusionPipeline,
+)
+from diffusers.schedulers.scheduling_utils import SCHEDULER_CONFIG_NAME
+from diffusers.utils import CONFIG_NAME
+from huggingface_hub import snapshot_download
+from transformers import CLIPFeatureExtractor, CLIPTokenizer
+from transformers.file_utils import add_start_docstrings
+
+import deepsparse
+from optimum.exporters.onnx import main_export
+from optimum.onnx.utils import _get_external_data_paths
+from optimum.pipelines.diffusers.pipeline_stable_diffusion import StableDiffusionPipelineMixin
+from optimum.utils import (
+    DIFFUSION_MODEL_TEXT_ENCODER_2_SUBFOLDER,
+    DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER,
+    DIFFUSION_MODEL_UNET_SUBFOLDER,
+    DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER,
+    DIFFUSION_MODEL_VAE_ENCODER_SUBFOLDER,
+)
+
+from .modeling_base import DeepSparseBaseModel
+
+
+ONNX_WEIGHTS_NAME = "model.onnx"
+os.environ["NM_DISABLE_BATCH_OVERRIDE"] = "1"
```

Comment:
move this within the compile function - we don't want to affect other code executing that might import this